lib/dry/cli.rb

```diff
@@ -135,23 +132,50 @@ def parse(command, arguments, names)
 
       result = Parser.call(command, arguments, prog_name)
 
-      if result.help?
-        out.puts Banner.call(command, prog_name)
-        exit(0)
-      end
+      return help(command, prog_name) if result.help?
 
-      if result.error?
-        err.puts(result.error)
-        exit(1)
-      end
+      return error(result) if result.error?
+
+      [command_builder(command), result.arguments]
+    end
 
-      [command.new, result.arguments]
+    # Prepares command instance and injects context
+    #
+    # @param command [Dry::CLI::Command]
```

Comment:
Shouldn't this reflect that it accepts a command class _or_ instance?

More generally: do we actually _want_ full yardocs for clearly internal, private helpers? /cc @jodosha for your thoughts on this. It feels like an unnecessary amount of ceremony/maintenance burden in this case...

Reply:
@timriley To me, the minimum set of API docs for private helpers is `@api private` and `@since`. The first one warns users to not use that method, the second one is for our own ease of maintenance.

On Hanami, we use to set `@since x.x.x` and before the release we change all the instances of that `x.x.x` with the actual version that is about to be released.

Reply:
Right. Maybe we could just shrink the comments down to that minimum set? Having full “documentation” for trivial, self-explanatory methods seems unnecessary to me, and it makes the class file noisier and overall more work to maintain. What do you think?